FitGoal/FitGoal/Singup&Singin/AuthenticationViewController.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  AuthenticationViewController.swift
+//  FitGoal
+//
+//  Created by Eliany Barrera on 3/4/20.
+//  Copyright © 2020 Eliany Barrera. All rights reserved.
+//
+
+import UIKit
+
+class AuthenticationViewController: UIViewController {
+    
+    private var authenticator: SocialMediaAuthenticator
+    
+    private let placeholder: UIActivityIndicatorView = {
+        let placeholder = UIActivityIndicatorView()
+        placeholder.color = #colorLiteral(red: 0.5568627715, green: 0.3529411852, blue: 0.9686274529, alpha: 1)
+        placeholder.style = .large
+        return placeholder
+    }()
+    
+    private var blurEffectView: UIVisualEffectView = {
+        let blurEffectView = UIVisualEffectView()
+        let blurEffect = UIBlurEffect(style: .extraLight)
+        blurEffectView.effect = blurEffect
+        return blurEffectView
+    }()
+    
+    private var loginStatus: LoginStatus? {
+        didSet {
+            DispatchQueue.main.async {
+                switch self.loginStatus {
+                case .loggedIn:
+                    let vc = HomeViewController(persistance: CoreDataPersistance())
+                    vc.modalPresentationStyle = .fullScreen
+                    vc.modalTransitionStyle = .crossDissolve
+                    self.present(vc, animated: true)
+                case .attempting:
+                    self.view.addSubview(self.placeholder)
+                    self.placeholder.center = self.view.center
```

Comment:
do you need this line? In case the `viewDidLayoutSubviews` function isn't called, you should mark the view with `setNeedsLayoutSubview`.